### PR TITLE
[ACM-6585]: Firewall and port requirements for bare metal and kubevirt

### PR DESCRIPTION
**Pull request by lahinson**
Related Jira: https://issues.redhat.com/browse/ACM-6585

### CODE REVIEW

**Comment by lahinson** [comment]
@zanetworker PTAL at the firewall and port requirements when you have a moment. Thank you!

File: clusters/hosted_control_planes/configure_hosted_bm.adoc [hunk 1]
```diff
@@ -44,6 +44,26 @@ oc get managedclusters local-cluster
...
+Ensure that you meet the firewall and port requirements so that ports can communicate between the management cluster, the control plane, and hosted clusters:
+
+* The `kube-apiserver` service runs on port 6443 by default and requires Ingress access for communication between the control plane components. 
+
+** If you use the `NodePort` publishing strategy, ensure that the node port that is assigned to the `kube-apiserver` service is exposed.
+** If you use MetalLB load balancing, allow Ingress access to the IP range that is used for load balancer IP addresses.
+
+* The `konnectivity` agent, which establishes a reverse tunnel to allow bi-directional communication on the hosted cluster, requires egress access to the cluster API server address on port 6443. With that egress access, the agent can reach the `kube-apiserver` service.
+
+** If the cluster API address is an internal IP address, allow access from the workload subnets to the IP address on port 6443.
```
Comment (zanetworker) [inline clusters/hosted_control_planes/configure_hosted_bm.adoc]:
```suggestion
** If the cluster API server address is an internal IP address, allow access from the workload subnets to the IP address on port 6443.
```

**Review by zanetworker — APPROVED** [review]
LGTM but I am biased :) 

It would be great if we can get more eyes on it, sent a request on slack.

File: clusters/hosted_control_planes/configure_hosted_bm.adoc (hunk 1 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_bm.adoc]: Good catch! Fixed in both files.

**Comment by bryan-cox** [comment]
This looks good from my perspective.

**Comment by zanetworker** [comment]
Based on feedback from the slack discussions, it seems we are missing a mention for the ignition-server. On the OpenShift docs, its port: `22623` https://access.redhat.com/solutions/4926401 but I can see its `9090` on OpenShift. But what we really want to add is for `NodePorts` related to `ignition-server` and `Oauth-server` there should be a firewall rule. I guess these would be the two most important port: @jparrill @bryan-cox  WDYT?

**Comment by jparrill** [comment]
> Based on feedback from the slack discussions, it seems we are missing a mention for the ignition-server. On the OpenShift docs, its port: `22623` https://access.redhat.com/solutions/4926401 but I can see its `9090` on OpenShift. But what we really want to add is for `NodePorts` related to `ignition-server` and `Oauth-server` there should be a firewall rule. I guess these would be the two most important port: @jparrill @bryan-cox WDYT?

Yes, I think in the same way. How could we help to articulate this in the doc?

File: clusters/hosted_control_planes/configure_hosted_bm.adoc [hunk 2]
```diff
@@ -44,6 +44,28 @@ oc get managedclusters local-cluster
...
+== Bare metal firewall and port requirements
+
+Ensure that you meet the firewall and port requirements so that ports can communicate between the management cluster, the control plane, and hosted clusters:
+
+* The `kube-apiserver` service runs on port 6443 by default and requires Ingress access for communication between the control plane components. 
+
+** If you use the `NodePort` publishing strategy, ensure that the node port that is assigned to the `kube-apiserver` service is exposed.
+** If you use MetalLB load balancing, allow Ingress access to the IP range that is used for load balancer IP addresses.
+
+* If you use the `NodePort` publishing strategy, use a firewall rule for the `ignition-server` and `Oauth-server` settings.
```
Comment (lahinson) [inline clusters/hosted_control_planes/configure_hosted_bm.adoc:61]: > Based on feedback from the slack discussions, it seems we are missing a mention for the ignition-server. On the OpenShift docs, its port: `22623` https://access.redhat.com/solutions/4926401 but I can see its `9090` on OpenShift. But what we really want to add is for `NodePorts` related to `ignition-server` and `Oauth-server` there should be a firewall rule. I guess these would be the two most important port: @jparrill @bryan-cox WDYT?

@zanetworker @jparrill @bryan-cox I added line 57 to mention that a firewall rule is needed for `ignition-server` and `Oauth-server`. PTAL and let me know if anything should change.

Comment (jparrill) [reply clusters/hosted_control_planes/configure_hosted_bm.adoc:61]: LGTM

**Comment by zanetworker** [comment]
lgtm

File: clusters/hosted_control_planes/configure_hosted_bm.adoc [hunk 3]
```diff
@@ -44,6 +44,28 @@ oc get managedclusters local-cluster
...
+
+* The `konnectivity` agent, which establishes a reverse tunnel to allow bi-directional communication on the hosted cluster, requires egress access to the cluster API server address on port 6443. With that egress access, the agent can reach the `kube-apiserver` service.
+
+** If the cluster API server address is an internal IP address, allow access from the workload subnets to the IP address on port 6443.
+** If the address is an external IP address, allow egress on 6443 to that external IP address from the nodes.
```
Comment (cadawson) [inline clusters/hosted_control_planes/configure_hosted_bm.adoc]:
```suggestion
** If the address is an external IP address, allow egress on port 6443 to that external IP address from the nodes.
```

File: clusters/hosted_control_planes/configure_hosted_bm.adoc [hunk 4]
```diff
@@ -44,6 +44,28 @@ oc get managedclusters local-cluster
...
 
+[#firewall-port-reqs-bare-metal]
+== Bare metal firewall and port requirements
+
+Ensure that you meet the firewall and port requirements so that ports can communicate between the management cluster, the control plane, and hosted clusters:
+
+* The `kube-apiserver` service runs on port 6443 by default and requires Ingress access for communication between the control plane components. 
+
+** If you use the `NodePort` publishing strategy, ensure that the node port that is assigned to the `kube-apiserver` service is exposed.
+** If you use MetalLB load balancing, allow Ingress access to the IP range that is used for load balancer IP addresses.
```
Comment (cadawson) [inline clusters/hosted_control_planes/configure_hosted_bm.adoc]: I noticed that Ingress is capitalized, while egress is not. Should those be consistent, or is there a reason for that?

File: clusters/hosted_control_planes/managing_hosted_kubevirt.adoc [hunk 5]
```diff
@@ -32,6 +32,28 @@ oc patch storageclass ocs-storagecluster-ceph-rbd -p '{"metadata": {"annotations
...
 
+[#firewall-port-reqs-kubevirt]
+== Firewall and port requirements
+
+Ensure that you meet the firewall and port requirements so that ports can communicate between the management cluster, the control plane, and hosted clusters:
+
+* The `kube-apiserver` service runs on port 6443 by default and requires Ingress access for communication between the control plane components. 
+
+** If you use the `NodePort` publishing strategy, ensure that the node port that is assigned to the `kube-apiserver` service is exposed.
+** If you use MetalLB load balancing, allow Ingress access to the IP range that is used for load balancer IP addresses.
```
Comment (cadawson) [inline clusters/hosted_control_planes/managing_hosted_kubevirt.adoc]: Same Ingress question.

File: clusters/hosted_control_planes/managing_hosted_kubevirt.adoc [hunk 6]
```diff
@@ -32,6 +32,28 @@ oc patch storageclass ocs-storagecluster-ceph-rbd -p '{"metadata": {"annotations
...
+
+* If you use the `NodePort` publishing strategy, use a firewall rule for the `ignition-server` and `Oauth-server` settings.
+
+* The `konnectivity` agent, which establishes a reverse tunnel to allow bi-directional communication on the hosted cluster, requires egress access to the cluster API server address on port 6443. With that egress access, the agent can reach the `kube-apiserver` service.
+
+** If the cluster API server address is an internal IP address, allow access from the workload subnets to the IP address on port 6443.
+** If the address is an external IP address, allow egress on 6443 to that external IP address from the nodes.
+
+* If you change the default port of 6443, adjust the rules to relfect that change.
```
Comment (cadawson) [inline clusters/hosted_control_planes/managing_hosted_kubevirt.adoc]:
```suggestion
* If you change the default port of 6443, adjust the rules to reflect that change.
```

**Review by cadawson — APPROVED** [review]
Looks good, Laura. I noted a couple of small things for consideration. Thanks!

File: clusters/hosted_control_planes/configure_hosted_bm.adoc (hunk 3 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_bm.adoc]: Good catch!

File: clusters/hosted_control_planes/configure_hosted_bm.adoc (hunk 4 above)
Comment (lahinson) [reply clusters/hosted_control_planes/configure_hosted_bm.adoc]: Good question! I thought the guidance from the Supplementary Style Guide was to always capitalize Ingress, but on further investigation, I learned that it's capitalized only in references to the Ingress object or the Ingress Controller. I updated the capitalization in this file and in any other files where "Ingress" is mentioned. Thanks!

**Pull request merged**